components/board/customlinux/board.go

```diff
@@ -0,0 +1,120 @@
+//go:build linux
+
+// Package customlinux implements a board running linux
+package customlinux
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"strconv"
+
+	"github.com/edaniels/golog"
+	"periph.io/x/host/v3"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/board/genericlinux"
+	"go.viam.com/rdk/resource"
+)
+
+func init() {
+	if _, err := host.Init(); err != nil {
+		golog.Global().Debugw("error initializing host", "error", err)
+	}
+
+	RegisterCustomBoard("customlinux")
+}
+
+// customLinuxBoard wraps the genericlinux board type so that both can implement their own Reconfigure function
+type customLinuxBoard struct {
+	*genericlinux.SysfsBoard
+}
+
+// RegisterCustomBoard registers a sysfs based board using the pin mappings.
+func RegisterCustomBoard(modelName string) {
+	resource.RegisterComponent(
+		board.API,
+		resource.DefaultModelFamily.WithModel(modelName),
+		resource.Registration[board.Board, *Config]{
+			Constructor: func(
```

Comment:
move all the logic in the constructor to a function that you can pass in delambdafied.